torch/csrc/utils/python_dispatch.cpp

```diff
@@ -345,6 +345,64 @@ void initDispatchBindings(PyObject* module) {
     return at::isTensorSubclassLike(tensor);
   });
 
+  m.def("_dispatch_total_keys", []() {
+    return (int64_t)c10::DispatchKey::EndOfFunctionalityKeys;
+  });
+  m.def("_dispatch_key_name", [](uint64_t dispatch_key) {
+    auto dt = (c10::DispatchKey)dispatch_key;
+    return c10::toString(dt);
+  });
+  m.def("_dispatch_num_backends", []() {
+    return c10::num_backends;
+  });
+
+  py::enum_<c10::DispatchKey>(m, "DispatchKey")
+    .value("Undefined", c10::DispatchKey::Undefined)
+    .value("Dense", c10::DispatchKey::Dense)
+    .value("BackendSelect", c10::DispatchKey::BackendSelect)
+    .value("CPU", c10::DispatchKey::CPU)
+    .value("CUDA", c10::DispatchKey::CUDA)
+    .value("AutocastCPU", c10::DispatchKey::AutocastCPU)
+    .value("AutocastCUDA", c10::DispatchKey::AutocastCUDA)
+    .value("AutogradCPU", c10::DispatchKey::AutogradCPU)
+    .value("ADInplaceOrView", c10::DispatchKey::ADInplaceOrView)
+    .value("AutogradCUDA", c10::DispatchKey::AutogradCUDA)
+    .value("PythonTLSSnapshot", c10::DispatchKey::PythonTLSSnapshot)
+    .value("Python", c10::DispatchKey::Python);
```

Comment:
The enum here is inconsistent with the prevailing convention for how dispatch keys are represented in Python, which is as strings that get parsed with `c10::parseDispatchKey` when they hit C++ (see torch/csrc/utils/python_dispatch.cpp) for examples. For consistency, it would be best to unify these. (An enum is fine, but the prevailing APIs should accept the enum, and that ought to be done in a separate PR.)

Reply:
Seems reasonable. Do you want me to fix that in this PR or a subsequent one?

Reply:
can be later